yt_dlp/extractor/yandexvideo.py

```diff
@@ -148,6 +148,40 @@ def _real_extract(self, url):
         }
 
 
+class YandexVideoPreviewIE(InfoExtractor):
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            yandex\.ru/video/preview(?:/?\?.*?filmId=|/)
+                        )
+                        (?P<id>\d+)
+                    '''
```

Comment:
proly not the best but this is more compact and what i prefer tbh
```suggestion
    _VALID_URL = r'https?://(?:www\.)?yandex\.ru/video/preview/(?:\?.*filmId=)?(?P<id>\d+)'
```